partiql-ast/src/main/java/org/partiql/ast/literal/Literal.java

```diff
@@ -0,0 +1,125 @@
+package org.partiql.ast.literal;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.ast.AstNode;
+import org.partiql.ast.AstVisitor;
+import org.partiql.ast.DataType;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Collection;
+import java.util.Collections;
+
+public abstract class Literal extends AstNode {
+    @NotNull
+    public abstract LiteralKind kind();
```

Comment:
Not exactly a fat-interface but similar pattern! We are doing a tagged-union, and we don't actually want a LiteralKind enum and the code (aka tag) should go directly on the Literal (like we did with DataType).

Reply:
I'll try using the code directly on the Literal. I had ran into some slightly repetitive + confusing code pattern when using a LiteralKind:

```
val litText = when (lit.kind().code()) {
    LiteralKind.NULL -> "NULL"
    LiteralKind.MISSING -> "MISSING"
    LiteralKind.BOOLEAN -> lit.booleanValue().toString()
    ...
    else -> error("Unsupported literal kind ${lit.kind()}")
}
```
^ `lit.kind().code()` rather than just `lit.code()`.

Reply:
Changed to be more like `DataType` in the lateset commit.